### ISSUE TITLE
Arrays?

### ISSUE THREAD

**Issue by xcthulhu** [drcode/ethereum-order-statistic-tree#3]
Apparently solidity has support for arrays now, so some of the `map(uint => _)` structures can be replaced.

**Comment by drcode** [comment]
I agree completely, I will leave this issue open until I can transition the relevant parts to arrays.
